themes/default/content/blog/examining-recommended-practices-code-placement-and-stacks/index.md

```diff
@@ -0,0 +1,148 @@
+* What about CI/CD? If an organization will use CI/CD, that may affect how you organize your code. (This is something we'll delve into more deeply later in this series.)
+
+It's worth stressing that the answers to these questions may change over time as organizations grow and their applications evolve. Addressing that change and its effects on an organization's Pulumi projects and stacks is something we'll be discussing later in this series.
+
+{{% notes type="info" %}}
+The discussion above is working from the assumption that Pulumi users will store their Pulumi code in a version control system, like Git. Strictly speaking, using a version control system isn't required for Pulumi to function (there is one exception we'll touch on later in this series when we discuss [Pulumi Deployments](https://www.pulumi.com/docs/intro/pulumi-service/deployments/)), but we **very strongly** recommend that all users use Pulumi in conjunction with version control.
+{{% /notes %}}
+
+In the case of Zephyr, their team felt like a monorepo approach (storing Pulumi code and application code together) was most appropriate for their specific requirements. Zephyr is a relatively small organization with a single team of developers that are responsible for managing both the cloud infrastructure as well as the development and deployment of the online store application. Zephyr's team also decicded to use a single Pulumi project---for now. (Be sure to follow this blog series to see how that evolves, and why!)
+
+### Addressing the need for multiple environments
+
+The second question Zephyr encountered is how to handle the need for multiple instances of their application. In addition to a production instance---which is the instance behind Zephyr's public-facing online store---Zephyr also felt they needed an environment for the developers to use in testing changes to the online store.
+
+This use case---needing to have multiple, separate instances of the infrastructure and applications created by a single Pulumi program---is exactly what [Pulumi stacks](https://www.pulumi.com/docs/intro/concepts/stack/) were designed to address. Each stack is a separate instance of the resources created by a Pulumi program within a project. Further, each stack has its own independent state, and each stack has its own configuration values. Stacks can be short-lived (meaning the associated resources are also short-lived), or stacks can be long-lived. Aside from the cloud resources created the Pulumi program, stacks are lightweight and simple to create or delete.
+
+<!--Are there other considerations that are involved in deciding whether or not to use stacks?-->
```

Comment:
Fundamentally that's it, at least when it comes to the single-repo approach we're outlining here. The need to be able to run multiple, independently configured instances of a program, typically mapped to shared environments (testing, staging, prod) and/or ad-hoc environments (developer stacks, long-running feature-branch stacks, etc.), is what usually drives this.

Reply:
Perfect, thanks.